src/modules/ExplorePage/Components/ExplorePage/ExplorePage.tsx

```diff
@@ -1,10 +1,230 @@
-import Footer from '../../../Shared/Components/Footer/Footer';
+import { Grid2 as Grid } from '@mui/material';
+import { useEffect, useState } from "react";
+import { axiosInstance, ROOMS_URLS } from "../../../../services/urls";
+import { Box, CircularProgress, Grid2, Typography } from "@mui/material";
+import NoData from "../../../Shared/Components/NoData/NoData";
+import roomImg from '../../../../assets/images/room.png';
+import UserPageTitle from '../../../Users-Portal/Component/UsersShared/UserPageTitle/UserPageTitle';
+import { useLocation } from 'react-router-dom';
+import dayjs from 'dayjs';
 
 export default function ExplorePage() {
-	return (
-		<>
-			<div>ExplorePage</div>
-			<Footer />
-		</>
-	);
+  const [availableRooms, setAvailableRooms] = useState<roomsdata[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  interface roomsdata {
+    roomNumber: string;
+    price: number;
+    capacity: number;
+    discount: number;
+    facilities: { name: string }[];
+    createdBy: string;
+    images: string[];
+    isBooked: boolean;
+    createdAt: string;
+    updatedAt: string;
+  }
+
+  interface availableRoomsResponse {
+    success: boolean;
+    message: string;
+    data: { rooms: roomsdata[]; totalCount: number };
+  }
+
+  const location = useLocation();
+  const { startDate, endDate } = location.state || {};
```

Comment:
We can use search params instead of location state. This will make the params in the url and shareable with other people.